### PR TITLE
Allow all users to see all campaigns

### PR DESCRIPTION
**Pull request by zzma**
@jeyrey @jddicki2 

Remove user dependency for fetching campaigns, groups, etc...

Also clean up model seeding magic numbers, and other small fixes.

### CODE REVIEW

**Comment by zzma** [comment]
I need to add some tests still, but feel free to provide feedback now

File: controllers/api.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func API_Campaigns(w http.ResponseWriter, r *http.Request) {
 			Logger.Println(err)
 		}
 		JSONResponse(w, cs, http.StatusOK)
-	//POST: Create a new campaign and return it as JSON
+		//POST: Create a new campaign and return it as JSON
```
Comment (jddicki2) [inline controllers/api.go]: Did you mean for this tab to get added.

**Comment by jddicki2** [comment]
Do we need to do anything about the changes failing the main gophish branch tests?

**Comment by zzma** [comment]
@jddicki2 - yeah I'll make sure it passes Travis before merging. It wasn't actually failing a test, it was failing to build only on golang 1.9 for some reason. I re-started the build and hopefully it was a Travis fluke.

File: controllers/api.go (hunk 1 above)
Comment (zzma) [reply controllers/api.go]: Nope - good catch. Fixed

**Comment by zzma** [comment]
Added some tests - this should be ready to go now. This PR adds a new development user "test" with pw "test" who should be able to see and edit the same campaigns and emails that the "admin" user can, and vice versa. 

There was a little bit of scope creep, fixing a few things that I saw along the way. 
1. Development seeding - refactored to remove magic numbers, fixed some specific fields, added a test
2. Testing - refactored `models_test.go` to use the same (and superior) testing framework `github.com/stretchr/testify` as the rest of the test suites, completely removed `gopkg.in/check.v1` dependency. Refactored tests to use `testing.json` for config. Also added a basic test for API endpoints. 
3. Misc - had to remove CSRF for testing environment because I couldn't figure out how to deterministically get the tests to account for it. Sometimes tests would pass, sometimes fail. I went down a sizable rabbit hole but wasn't able to figure it out, so I left a TODO instead. Also made minor changes to email error logging and added a couple utility functions to `user.go`

**Pull request merged**